### PR TITLE
Clean base implementation for BlobStorageSpaces

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/resources/component-biz.conf [hunk 1]
```diff
@@ -581,7 +581,19 @@ storage {
...
+    # Contains settings for the blob storage system.
     layer2 {
+        # Controls the conversion settings used by the BlobStorageSpace to generate variants of a blob.
+        conversion {
+
+            # Determines if conversions are enabled on this node
+            enabled = true
+
+            # Contains a list of hostnames or IP addresses to which a conversion can be delegated by
+            # forwaring a request.
```
Comment (sabieber) [inline src/main/resources/component-biz.conf]: forwaring -> forwarding

File: src/main/resources/component-biz.conf [hunk 2]
```diff
@@ -908,6 +920,20 @@ async.executor {
         poolSize = 10
         queueLength = 0
     }
+
+    # This executor collects all reuquests which are served via the storage framework. These requests might
```
Comment (sabieber) [inline src/main/resources/component-biz.conf]: reuquests -> requests

File: src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java [hunk 3]
```diff
@@ -161,12 +273,241 @@ public Directory getRoot(String tenantId) {
...
+     * Performs a rollback for the given blob.
+     * <p>
+     * This is the <tt>rollback</tt> callback for all optimistic locking algorithms concerning blobs.
+     *
+     * @param blob the blob to delete
+     */
+    protected abstract void rollbackBlob(B blob);
+
+    /**
+     * Tries to find or create child directory with the given name.
```
Comment (meg1502) [inline src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java]:
```suggestion
     * Tries to find or create the child directory with the given name.
```

File: src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java [hunk 4]
```diff
@@ -161,12 +273,241 @@ public Directory getRoot(String tenantId) {
...
+     *
+     * @param parent         the parent directory to search in
+     * @param childName      the name of the directory
+     * @param childDirectory the directory to check
+     * @return <tt>true</tt> if there is only one child directory with this name in the given parent directory
+     */
+    protected abstract boolean isChildDirectoryUnique(D parent, String childName, D childDirectory);
+
+    /**
+     * Tries to find or create child blob with the given name.
```
Comment (meg1502) [inline src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java]:
```suggestion
     * Tries to find or create the child blob with the given name.
```

**Pull request merged**